snuba/datasets/entities/discover.py

```diff
@@ -348,18 +335,30 @@ def __init__(self) -> None:
         self.__transactions_columns = TRANSACTIONS_COLUMNS
 
         events_storage = get_storage(StorageKey.EVENTS)
-        events_ro_storage = get_storage(StorageKey.EVENTS_RO)
-        transactions_storage = get_storage(StorageKey.TRANSACTIONS)
 
         super().__init__(
-            storages=[events_storage, transactions_storage],
-            query_plan_builder=SelectedStorageQueryPlanBuilder(
-                selector=DiscoverQueryStorageSelector(
-                    events_table=events_storage,
-                    events_ro_table=events_ro_storage,
-                    abstract_events_columns=self.__events_columns,
-                    transactions_table=transactions_storage,
-                    abstract_transactions_columns=self.__transactions_columns,
+            storages=[events_storage],
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=events_storage,
```

Comment:
Shouldn't the discover entity use a selected storage query plan builder to switch between sentry_dist and sentry_dist_ro just like the events entity ?

Reply:
I decided to remove this from the base Discover entity since this table will only use the merge table in future (only when the events entity is selected will we switch between errors_dist and errors_dist_ro instead of events_dist and events_dist_ro).

Reply:
Right, but isn't this entity used today? This change would make it stop using sentry_dist_ro.

Reply:
Yes exactly. This PR drops that optimization (only for the Discover entity though) since that optimization won't exist on the Discover entity going forward. I'm happy to add it back temporarily if you feel like we should try and keep it around until we create the merge table for some reason.

Reply:
I think we should keep it now. It helps a lot on heavy queries (typical of discover)  to not add more load on the nodes that do ingestion and not to be impacted by the ingestion load.